File: python/binary_search/test_binary_search.py
import unittest
from binary_search import binary_search

class BinarySearchTest(unittest.TestCase):
    def test_binary_search(self):
        test_list = [1,3,9,11,15,19,29]
        self.assertEqual(binary_search(test_list, 25), -1)
        self.assertEqual(binary_search(test_list, 15), 4)
        self.assertEqual(binary_search(test_list, 30), -1)
        self.assertEqual(binary_search(test_list, 0), -1)

if __name__ == '__main__':
    unittest.main()

File: ruby/binary_search/test_binary_search.rb
require_relative 'binary_search'
require 'test/unit'

class TestBinarySearch < Test::Unit::TestCase
  def test_binary_search
    test_list = [1,3,9,11,15,19,29]
    assert_equal(-1, binary_search(test_list, 25))
    assert_equal(4, binary_search(test_list, 15))
    assert_equal(-1, binary_search(test_list, 30))
    assert_equal(-1, binary_search(test_list, 0))
    assert_equal(1, binary_search(test_list, 3))
  end
end
